### PR TITLE
Resolves #21 - Setup Redux

### LINKED ISSUE #21: Setup Redux
### Description

We need state management up and running

### How done looks like:

- Be able to start creating actions/reducers and hooking them up to components without needing set-up

- [ ] redux-thunk
- [ ] redux-dev-tool

### Ramp up tools
- https://auth0.com/blog/secure-your-react-and-redux-app-with-jwt-authentication/
- https://redux.js.org/

### PR DESCRIPTION
**Pull request by andela-ksolomon**
Resolves #21 

#### Description
Setup react redux for managing application state and also implement the redux dev tools libraries

#### Fix
- Add redux, react-redux library
- Add store and reducers files
- Setup redux dev tools

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -41,17 +41,6 @@
     "semi": [
       "error",
       "always"
-    ],
```
Comment (john555) [inline .eslintrc.json]: What is the purpose of removing this configuration?

File: package.json [hunk 2]
```diff
@@ -46,6 +46,7 @@
     "enzyme": "^3.3.0",
     "enzyme-adapter-react-16": "^1.1.1",
     "eslint-loader": "^1.9.0",
+    "eslint-plugin-jsx-a11y": "^6.0.3",
```
Comment (john555) [inline package.json]: What is the purpose of adding this package?

File: .eslintrc.json (hunk 1 above)
Comment (john555) [inline .eslintrc.json]: We need this configuration. It prevents us from getting this error.
```
[eslint] The href attribute is required on an anchor. Provide a valid, navigable address as the href value. (jsx-a11y/anchor-is-valid)
(JSX attribute) to: string
```

File: package.json (hunk 2 above)
Comment (john555) [inline package.json]: This is a peer dependency of `eslint-config-airbnb` and will be installed automatically. What was the motivation behind explicitly adding it here? Any advantages?

Comment (andela-ksolomon) [reply package.json]: I got an error initially, `ERROR in ./src/index.jsx
Module build failed: Error: Failed to load plugin jsx-a11y: Cannot find module 'eslint-plugin-jsx-a11y'
Referenced from: /Users/andeladeveloper/Desktop/andela-societies-frontend/node_modules/eslint-config-airbnb/index.js`

File: .eslintrc.json (hunk 1 above)
Comment (andela-ksolomon) [reply .eslintrc.json]: Okay, I see. Thanks

File: package.json (hunk 2 above)
Comment (andela-ksolomon) [reply package.json]: the configuration for  `jsx-a11y/anchor-is-valid` requires this

**Comment by john555** [comment]
LGTM. Kindly update the branch so that I can merge.

**Pull request merged**